src/libcore/macros.rs

```diff
@@ -145,11 +145,11 @@ macro_rules! assert_ne {
 /// # Uses
 ///
 /// Unlike [`assert!`], `debug_assert!` statements are only enabled in non
-/// optimized builds by default. An optimized build will omit all
+/// optimized builds by default. An optimized build will not execute
 /// `debug_assert!` statements unless `-C debug-assertions` is passed to the
 /// compiler. This makes `debug_assert!` useful for checks that are too
 /// expensive to be present in a release build but may be helpful during
-/// development.
+/// development. `debug_assert!` statements are always type checked.
```

Comment:
```suggestion
/// development. The result of expanding `debug_assert!` is always type checked.
```
(It produces an expression, not a statement.)